backend/utils/validation.py

```
@@ -291,3 +359,55 @@ def validate_schema(data, schema_name):
+            end_date = datetime.fromisoformat(data['end_date'])
+
+            if end_date < start_date:
+                raise ValidationError("End date cannot be before start date")
+
+            # Check if dates are too far in the future (more than 1 year)
+            now = datetime.now()
+            if start_date > now.replace(year=now.year + 1):
+                raise ValidationError("Start date cannot be more than 1 year in the future")
+
+        except ValueError as e:
+            raise ValidationError(f"Invalid date format: {str(e)}")
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)